src/desktop/apps/authentication/helpers.ts

```diff
@@ -112,19 +118,55 @@ export const setCookies = options => {
   }
 }
 
-export const getRedirect = type => {
+export async function apiAuthWithRedirectUrl(
+  response: any,
+  redirectPath: URL
+): Promise<URL> {
+  const redirectUrl = sd.APP_URL + redirectPath.pathname
+  const accessToken = (response.user || {}).accessToken
+  const appRedirectURL = new URL(redirectUrl)
+
+  // There isn't an access token when we don't have a valid session, for example,
+  // when the user is resetting their password.
+  if (!accessToken) return appRedirectURL
+
+  try {
+    const tokenResponse = await fetch(sd.API_URL + "/api/v1/me/trust_token", {
+      method: "POST",
+      headers: { "X-Access-Token": accessToken },
+    })
+
+    if (tokenResponse.ok) {
+      const responseBody = await tokenResponse.json()
+      const trustToken = responseBody["trust_token"]
+
+      return new URL(
+        `${
+          sd.API_URL
+        }/users/sign_in?trust_token=${trustToken}&redirect_uri=${appRedirectURL.toString()}`
```

Comment:
Should we escape these query param values? Perhaps using `encodeURIComponent`?

It looks like we already use the `querystring` package elsewhere to encode params safely:

https://github.com/artsy/force/blob/33ece666e9df1cf32f9c6cbe7914811815a860cc/src/desktop/components/filter2/router/index.coffee#L1-L23

I'm also wondering if we'll run into browser compatibility issues using the URL API on some browsers: https://caniuse.com/#feat=url

Reply:
Good call on the URL encoding, addressed in 21ddbc322.

Re browser compatibility, here's a look that shows our current understanding of [session creations by browser type](https://artsy.looker.com/explore/analytics/sessions?qid=jTYcILwrr6IWMIN6SrvGHc&toggle=pik).

While IE doesn't support any this feature at any version, there's also only 6 IE 7 (??) sessions of out 590K. Moreover, we systematically [block IE anyways](https://github.com/artsy/force/blob/master/src/lib/middleware/unsupportedBrowser.ts#L22).

<img width="1552" alt="Screen Shot 2019-08-07 at 7 21 55 PM" src="https://user-images.githubusercontent.com/1561546/62664217-bd867400-b948-11e9-89a7-014147094711.png">

--

Generally, I'm pretty unfamiliar with our strategy with respect to browser support, so would definitely be curious to get others take here!

Reply:
We don't support anything less than modern IE (Edge)